src/components/Releases/Releases.vue

```diff
@@ -130,35 +131,46 @@ export default {
       this.$store.dispatch('setReleaseFinalDate', this.releases[this.releaseIndex].final_date);
     },
 
-    getReleases() {
+    async getReleases() {
       const headers = { Authorization: this.token };
 
-      HTTP.get(`projects/${this.projectId}/releases`, { headers })
-        .then((response) => {
+      try {
+        let response = await HTTP.get(`projects/${this.projectId}/releases`, { headers });
+        this.releases = response.data;
+        if(this.releases.length !== 0) {
           this.isLoaded = true;
-          this.releases = response.data;
-
+          this.setReleaseId();
+          this.getSprints();
           this.setAmountOfReleases();
-        })
-        .catch((e) => {
-          this.errors.push(e);
-        });
+          return;
+        }
+        
+        this.$store.dispatch('setRelease', null);
+        this.setAmountOfReleases();
+      }
+      catch(e) {
+        this.errors.push(e);
+      }
     },
 
-    getSprints() {
+    async getSprints() {
       const headers = { Authorization: this.token };
 
-      HTTP.get(`releases/${this.releaseId}/sprints`, { headers })
-        .then((response) => {
-          if (response.data !== 0) {
-            this.sprints = response.data;
-          } else {
-            this.sprint = 0;
-          }
-        })
-        .catch((e) => {
-          this.errors.push(e);
-        });
+      if(this.isReleasesEmpty()) {
+        return;
+      }
+
+      try {
+        let response = await HTTP.get(`releases/${this.releaseId}/sprints`, { headers });
+        if(response.data.length !== 0) {
+          this.sprints = response.data;
+          return;
+        }
+        this.sprints = [];
```

Comment:
Should this have a return after this line too? Why only inside the `if` condition?

Reply:
This if condition guarantees the returning of sprints not empty. If there is no sprint, I set explicitly that the `this.sprints` is an empty list, without it, the code catches the previous sprint wrongly.